### PR TITLE
5213: refactor search and select field to have same behavior and work…

### PR DESCRIPTION
**Pull request by transitorytammy**
… with groupedContent

screen shot 2019-01-25 at 2 40 09 pm
screen shot 2019-01-25 at 2 39 59 pm

### CODE REVIEW

File: addon/components/bourbon-select-field.js [hunk 1]
```diff
@@ -44,39 +45,46 @@ export default Component.extend(SelectMixin, {
...
-          this.setLabel(searchValue);
-          this.setValue(searchValue);
-        } else {
-          this.setLabel(value);
-          this.setValue(value);
-        }
+        this.setLabel(value);
+        this.setValue(value);
       }
-      return value;
```
Comment (lawrencegetfeedback) [inline addon/components/bourbon-select-field.js]: I think custom setters always need to return something.

File: addon/components/bourbon-search-select-field.js [hunk 2]
```diff
@@ -53,88 +56,39 @@ export default Component.extend(SelectMixin, {
...
-  },
-
-  resetOptions() {
-    this.send('updateSearchSelection');
-    this.set('showDropdown', false);
-    this.set('activeOption', null);
-    document.activeElement.blur();
+  keyDown(e) {
+    this.moveUpDown(e);
+    if (e.keyCode === 13) {
```
Comment (lawrencegetfeedback) [inline addon/components/bourbon-search-select-field.js]: could you add a comment with the corresponding key for keyCode 13?

File: addon/components/bourbon-select-field.js [hunk 3]
```diff
@@ -44,39 +45,46 @@ export default Component.extend(SelectMixin, {
...
 
   mouseDown() {
+    this.set('activeOption', null);
     this.set('showList', !this.get('showList'));
   },
 
   findValueObject(valueString) {
-    for (var optIndex in this.get('content')) {
-      if (this.get('content')[optIndex].value === valueString) {
+    for (var optIndex of this.get('content')) {
```
Comment (lawrencegetfeedback) [inline addon/components/bourbon-select-field.js]: did the type of `this.get('content')` change? not sure why this change from `in` to `of`

File: addon/mixins/select.js [hunk 4]
```diff
@@ -26,14 +26,61 @@ export default Mixin.create({
...
+  moveUpDown(e) {
+    let el = $(e.currentTarget);
+
+    let list = el.find('.BourbonSelectField-menu');
+    let allOptions = el.find(
+      '.BourbonSelectField-menu .BourbonSelectField-option'
+    );
+    let numOptions = allOptions.length;
+
+    if (e.keyCode === 40) {
```
Comment (lawrencegetfeedback) [inline addon/mixins/select.js]: also some comments for these keyCodes

File: addon/templates/components/bourbon-select-field.hbs [hunk 5]
```diff
@@ -21,11 +21,14 @@
...
   {{#if groupedContent}}
-    {{#each content as |group|}}
-      <div class="BourbonSelectField-optionGroupName">{{group.label}}</div>
-      {{#each group.items as |option index|}}
-        {{bourbon-select-field-option index=index tagName="div" content=option selection=selection labelPath=optionLabelPath valuePath=optionValuePath enabledPath=optionEnabledPath disabled=option.disabled target=target role="listitem"}}
-      {{/each}}
+    {{#each content as |group index|}}
+      <div class="BourbonSelectField-optionGroup" index={{index}}>
+        <div class="BourbonSelectField-optionGroupName">{{group.label}}</div>
+        {{#each group.items as |option index|}}
```
Comment (lawrencegetfeedback) [inline addon/templates/components/bourbon-select-field.hbs]: We probably should rename these `index` variables to `outerIndex` and `innerIndex` or something, as not to conflate the two in these nested loops.

File: addon/components/bourbon-select-field.js (hunk 1 above)
Comment (transitorytammy) [reply addon/components/bourbon-select-field.js]: Ok. I added this back.  :)

File: addon/components/bourbon-search-select-field.js (hunk 2 above)
Comment (transitorytammy) [reply addon/components/bourbon-search-select-field.js]: done

File: addon/mixins/select.js (hunk 4 above)
Comment (transitorytammy) [reply addon/mixins/select.js]: done :)

File: addon/templates/components/bourbon-select-field.hbs (hunk 5 above)
Comment (transitorytammy) [reply addon/templates/components/bourbon-select-field.hbs]: done

File: addon/components/bourbon-select-field.js (hunk 3 above)
Comment (transitorytammy) [reply addon/components/bourbon-select-field.js]: not sure what happened here.  i switched it back to in.

File: addon/mixins/select.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import Mixin from "@ember/object/mixin";
 
 export default Mixin.create({
   setLabel(value) {
-    let checkValue = value ? value : this.get('value')
+    let checkValue = this.getCheckValue(value);
```
Comment (lawrencegetfeedback) [inline addon/mixins/select.js]: Could you explain what's going on around this `checkValue` and `getCheckValue` code? Just a little hard to follow on a Monday morning haha.

Comment (transitorytammy) [reply addon/mixins/select.js]: I had to add this because when passing in "groupedContent" it was changing the format of what I was expecting so, I needed change whether I was using `value` or `this.get('value')`.

**Comment by transitorytammy** [comment]
1 day in January
